File: 1. Two Sum/two_sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        n=len(nums)
        mp={}
        for i in range(n):
            res=target-nums[i]
            if res in mp:
                return [mp[res],i]
            else:
                mp[nums[i]]=i
 

File: 1. Two Sum/two_sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        int n = nums.size();
        vector<int> ans;
        map<int,int> mp;
        for (int i = 0; i<n; i++) mp[nums[i]] = -1;
        for (int i = 0; i<n; i++){
            if (mp[nums[i]] != -1) ans={mp[nums[i]],i};
            mp[target-nums[i]] = i;
        }
        return ans;
    }
};
